src/eei.cpp

```diff
@@ -974,11 +977,11 @@ inline int64_t maxCallGas(int64_t gas) {
     ensureCondition(safeLoadUint128(balance) >= safeLoadUint128(value), OutOfGasException, "Out of gas.");
   }
 
-  uint64_t EthereumInterface::safeLoadUint128(evmc_uint256be const& value)
+  intx::uint128 EthereumInterface::safeLoadUint128(evmc_uint256be const& value)
```

Comment:
You know that GCC / clang have `unsigned __int128`?

Reply:
@chfast would have been cool if you could have mentioned that when the original issue was created :)

But according to https://gcc.gnu.org/onlinedocs/gcc/_005f_005fint128.html :
> As an extension the integer scalar type __int128 is supported for targets which have an integer mode wide enough to hold 128 bits. 
> There is no support in GCC for expressing an integer constant of type __int128 for targets with long long integer less than 128 bits wide.

Reply:
So we can assume it works for x86-64 but not much else until we investigate

Reply:
I wasn't aware that 128-bit is the max precision you need.

Anyway, the type is available on 64-bit targets. Because you don't support MSVC at the moment it can be used as a temporary solution.